artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/ActiveMQServerControlImpl.java

```diff
@@ -4535,8 +4537,8 @@ private PropertiesLoginModuleConfigurator getPropertiesLoginModuleConfigurator()
       if (configurationUrl == null) {
          throw ActiveMQMessageBundle.BUNDLE.failedToLocateConfigURL();
       }
-      String path = configurationUrl.getPath();
-      return new PropertiesLoginModuleConfigurator(getSecurityDomain(), path.substring(0, path.lastIndexOf("/")));
+      String path = Paths.get(configurationUrl.toURI()).toString();
```

Comment:
The doc for Paths indicates to use Path.of(..) instead.

Reply:
Fixed.